scripts/deploy/github_steps.rb

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env ruby
+
+require 'open3'
+require 'octokit'
+
+require_relative 'common'
+
+def ensure_clean_repo()
+  repo_dir = File.basename(Dir.getwd)
+  if repo_dir != "stripe-android"
+    abort("You must run this script from 'stripe-android'.")
+  end
+
+  stdout, stderr, status = Open3.capture3("git status --porcelain")
+
+  if !stdout.empty?
+    abort("You must have a clean working directory to continue.")
+  end
+end
+
+def pull_latest()
+    execute_or_fail("git checkout #{@branch}")
+    execute_or_fail("git pull")
+end
+
+def switch_to_release_branch()
+    # Using the -B flag will create a new branch if it doesn't exist, or will reset the existing
+    # branch.
+    execute_or_fail("git checkout -B #{release_branch}")
+end
+
+def delete_release_branch()
+    execute_or_fail("git push origin --delete #{release_branch}")
+    execute_or_fail("git branch -d #{release_branch}")
+end
+
+def create_version_bump_pr()
+    begin
+        execute_or_fail("git add README.md")
+        execute_or_fail("git add gradle.properties")
+        execute_or_fail("git add stripe-core/src/main/java/com/stripe/android/core/version/StripeSdkVersion.kt")
+        execute_or_fail("git commit -m \"Bump version to #{@version}\"")
+        execute_or_fail("git push -u origin")
+    rescue
+        # Undo all of the above if any of the steps fail.
+        execute_or_fail("git reset HEAD~")
+        # Don't continue if any of the above failed.
+        raise
```

Comment:
Can we raise the issue that created this, so we have a better understanding of what failed.

Reply:
I updated this so that we wouldn't raise an issue if `git reset HEAD~` failed. Otherwise, my understanding is that calling `raise` with no params will raise the last thrown exception, so it will raise the issue that created it. I also verified this locally.